### DISCUSSION TITLE
Why `hash` of `pymatgen.analysis.reaction_calculator.BalancedReaction` is hardcoded as int `7`?

### DISCUSSION THREAD

**Discussion by DanielYang59** [materialsproject/pymatgen#3673] [Q&A] [answered]
I noticed the following definition while working with `BalancedReaction`, just wondering why this magic number `7`?

https://github.com/materialsproject/pymatgen/blob/a5a48874a3baeeb95f1fecc206b62a00554ab992/pymatgen/analysis/reaction_calculator.py#L173-L174

This would make `BalancedReaction` "hashable", but then all instances would have the same hash?

**Comment by janosh** [comment, marked as answer]
beats me. i assume there's no good reason for this particular value. we could probably change it to

```py
 def __hash__(self) -> int: 
    return hash((self.reactants_coeffs, self.products_coeffs))
```

or remove it altogether

**Reply by DanielYang59** [reply]
That makes sense. But I guess it might be better to provide a hash in case someone need it for some reason.

~~With that said, maybe `hash((self.coeffs, self.reactants, self.products))` to take into account not only coefficients but species themselves? I could try to implement this change if so.~~ 
Sorry I thought `reactants_coeffs` is just the coefficients, you are right.

**Reply by janosh** [reply]
> I could try to implement this change if so.

go ahead! also, maybe add a comment that the need for a hash method is unclear (unless @mkhorton @shyuep know why?)
